auth/client.go

```diff
@@ -1,12 +1,35 @@
 package auth
 
 import (
+	"errors"
 	"io/ioutil"
 	"net/http"
+	"time"
 )
 
-// Returns new secret request
-func NewClient(teamId, clientId, keyId, secretKeyPath string) (*Request, error) {
+var InvalidSecretFileMsg = "please specify secret key file path"
+
+// Returns new secret request with defult client
+func New(teamId, clientId, keyId, secretKeyPath string) (*Request, error) {
+
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	return createRequest(teamId, clientId, keyId, secretKeyPath, client)
+}
+
+// Returns new secret request with given client
+func NewWithClient(client httpClient, teamId, clientId, keyId, secretKeyPath string) (*Request, error) {
```

Comment:
What's new with Client?

Reply:
New auth initialization with http client

Reply:
I think it can be renamed to some more meaningful method name.

Reply:
As of me it can be WithDefaultClient/WithCustomClient.